### PR TITLE
consistent ally-neutral-enemy

### PR DESCRIPTION
**Pull request by wwongkamjan**
instead of having `self.stance.stance[power][power] > 0` everywhere in SOA, we change to store

```
self.allies
self.foes
self.neutral
```

This reflects in every function related to the stance vector

### CODE REVIEW

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 1]
```diff
@@ -551,13 +552,28 @@ def __call__(self, rcvd_messages: List[Tuple[int, Message]]):
...
+
+
             # filter out aggressive orders to allies
             yield self.replace_aggressive_order_to_allies()
 
             # generate messages for FCT sharing info orders
             opps = list(powers.keys()).copy()
             opps.remove(self.power_name) # list of opposing powers
-            msgs_data = yield self.gen_messages(orders_data.get_list_of_orders())
+            msgs_data = yield self.gen_messages(orders_data.get_list_of_orders(), msgs_data)
```
Comment (wwongkamjan) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:576]: This is to add more flexibility to `gen_message` It should not create a new msgs_data = MessageData but use the existing one that we already have. Oh though this is (functionally) not in used anymore

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 2]
```diff
@@ -88,16 +91,15 @@ async def gen_pos_stance_messages(
                 )
             )
             if str(orders_decided) != "FCT ()":
-                for pow in self.stance.stance[self.power_name]:
+                for pow in self.allies:
                     if (
                         pow != self.power_name
-                        and self.stance.stance[self.power_name][pow] > 0
+                        and pow in self.allies
```
Comment (kartik2112) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:97]: redundant line

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 3]
```diff
@@ -551,13 +552,29 @@ def __call__(self, rcvd_messages: List[Tuple[int, Message]]):
...
-            yield self.replace_aggressive_order_to_allies()
+
+            # fmt: off
+
+            self.allies = [pow for pow in powers if (pow != self.power_name and powers[pow] >= self.ally_threshold)]
+            self.foes = [pow for pow in powers if (pow != self.power_name and powers[pow] <= self.enemy_threshold)]
+            self.neutral = [pow for pow in powers if (pow != self.power_name and powers[pow] > self.enemy_threshold and powers[pow] < self.ally_threshold)]
+
+            # GLOBAL message and filter aggressive moves to allies are disabled in S1901M
+            if self.game.get_current_phase()!='S1901M':
```
Comment (kartik2112) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:563]: Umm I think we are supposed to send stance messages to GLOBAL but in first phase, just claim all are NEUTRAL

**Review by kartik2112 — CHANGES_REQUESTED** [review]

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 3 above)
Comment (wwongkamjan) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:563]: oh I saw jbg said "i.e. don't print anything"

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 2 above)
Comment (wwongkamjan) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:97]: oh I just changed according to the prev version but yeah let's remove the redundant

Comment (wwongkamjan) [reply src/baseline_bots/bots/smart_order_accepter_bot.py]: done

**Review by kartik2112 — APPROVED** [review]

**Pull request merged**